parquet-tools/src/test/java/org/apache/parquet/tools/command/TestPruneColumnsCommand.java

```diff
@@ -0,0 +1,258 @@
+    String cargs[] = {inputFile, outputFile, "Links"};
+    executeCommandLine(cargs);
+
+    // Verify the schema are not changed for the columns not pruned
+    ParquetMetadata pmd = ParquetFileReader.readFooter(conf, new Path(outputFile), ParquetMetadataConverter.NO_FILTER);
+    MessageType schema = pmd.getFileMetaData().getSchema();
+    List<Type> fields = schema.getFields();
+    assertEquals(fields.size(), 3);
+    assertEquals(fields.get(0).getName(), "DocId");
+    assertEquals(fields.get(1).getName(), "Name");
+    assertEquals(fields.get(2).getName(), "Gender");
+
+    // Verify the data are not changed for the columns not pruned
+    List<String> prunePaths = Arrays.asList("Links");
+    validateColumns(inputFile, prunePaths);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNotExistsNestedColumn() throws Exception {
+    // Create Parquet file
+    String inputFile = createParquetFile();
+    String outputFile = getParquetFileName();
+    String cargs[] = {inputFile, outputFile, "Links.Not_exists"};
+    executeCommandLine(cargs);
+  }
+
+  private void executeCommandLine(String[] cargs) throws Exception {
+    CommandLineParser parser = new PosixParser();
+    CommandLine cmd = parser.parse(new Options(), cargs, command.supportsExtraArgs());
+    command.execute(cmd);
+  }
+
+  private void validateColumns(String inputFile, List<String> prunePaths) throws IOException {
+    ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), new Path(inputFile)).withConf(conf).build();
+    for (int i = 0; i < numRecord; i++) {
+      Group group = reader.read();
+      if (!prunePaths.contains("DocId")) {
+        assertEquals(1l, group.getLong("DocId", 0));
+      }
+      if (!prunePaths.contains("Name")) {
+        assertEquals("foo", group.getBinary("Name", 0).toStringUsingUTF8());
+      }
+      if (!prunePaths.contains("Gender")) {
+        assertEquals("male", group.getBinary("Gender", 0).toStringUsingUTF8());
+      }
+      if (!prunePaths.contains("Links")) {
+        Group subGroup = group.getGroup("Links", 0);
+        if (!prunePaths.contains("Links.Backward")) {
+          assertEquals(2l, subGroup.getLong("Backward", 0));
+        }
+        if (!prunePaths.contains("Links.Forward")) {
+          assertEquals(3l, subGroup.getLong("Forward", 0));
+        }
+      }
+    }
+    reader.close();
+  }
+
+  private String createParquetFile() throws IOException {
+    MessageType schema = new MessageType("schema",
+      new PrimitiveType(REQUIRED, INT64, "DocId"),
+      new PrimitiveType(REQUIRED, BINARY, "Name"),
+      new PrimitiveType(REQUIRED, BINARY, "Gender"),
+      new GroupType(OPTIONAL, "Links",
+        new PrimitiveType(REPEATED, INT64, "Backward"),
+        new PrimitiveType(REPEATED, INT64, "Forward")));
+
+    conf.set(GroupWriteSupport.PARQUET_EXAMPLE_SCHEMA, schema.toString());
+
+    Path file = new Path(root, + rand.nextInt(100000) + "input.parquet");
```

Comment:
I would prefer using creating the file in a temporary space instead. (See e.g. `java.nio.file.Files.createTempFile`)
Creating a file int the hard-coded `"target/"` relative path is a bit error prone. What happens if the `mvn` command is executed from outside of the project structure?

Reply:
Fixed